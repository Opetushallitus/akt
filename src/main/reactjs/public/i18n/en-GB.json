{
  "akt": {
    "header": {
      "title": "AKT",
      "lang": {
        "fi": "Suomeksi",
        "sv": "PÃ¥ Svenska",
        "en": "In English"
      }
    }
  }
}
