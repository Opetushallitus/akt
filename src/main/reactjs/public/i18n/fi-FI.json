{
  "akt": {
    "component": {
      "header": {
        "lang": {
          "fi": "Suomeksi",
          "sv": "På Svenska",
          "en": "In English"
        },
        "logo": {
          "alt": "Etusivu"
        },
        "clerk": {
          "backToOph": "Takaisin opintopolkuun",
          "logOut": "Kirjaudu ulos",
          "navLinks": {
            "register": "Rekisteri",
            "certificateCreation": "Todistuksen tekeminen",
            "meetingDates": "Kokouspäivät",
            "tabsLabel": "Virkailijan URL-osoitteet"
          }
        }
      },
      "table": {
        "title": "Hakutulokset",
        "pagination": {
          "rowsPerPage": "Näytä sivulla"
        },
        "selectedItems": "valittu"
      },
      "footer": {
        "links": {
          "opintopolku": {
            "text": "Opintopolku",
            "link": "https://opintopolku.fi/konfo/fi"
          },
          "accessibility": {
            "text": "Saavutettavuusseloste",
            "link": "https://www.oph.fi/fi/tietoa-meista/opetushallituksen-saavutettavuusseloste"
          },
          "aktHomepage": {
            "text": "Auktorisoidun kääntäjän tutkintojärjestelmä",
            "link": "https://www.oph.fi/fi/koulutus-ja-tutkinnot/kieli-ja-kaantajatutkinnot/auktorisoidun-kaantajan-tutkinto"
          }
        },
        "address": {
          "name": "Opetushallitus",
          "street": "Hakaniemenranta 6",
          "zipCity": "PL 380, 00531 Helsinki",
          "phone": {
            "title": "Puhelin",
            "number": "+358 29 533 1000"
          }
        },
        "logo": {
          "alt": "Etusivu"
        }
      },
      "publicTranslatorFilters": {
        "languagePair": {
          "title": "Haku kielittäin",
          "fromPlaceholder": "Mistä *",
          "fromHelperText": "Valitse kieli",
          "toPlaceholder": "Mihin *",
          "toHelperText": "Valitse kieli"
        },
        "name": {
          "title": "Haku nimellä",
          "placeholder": "Nimi"
        },
        "town": {
          "title": "Haku asuinkunnalla",
          "placeholder": "Asuinkunta"
        },
        "buttons": {
          "empty": "Tyhjennä valinnat",
          "search": "Hae",
          "newSearch": "Tee uusi haku"
        },
        "toasts": {
          "selectLanguagePair": "Valitse kielipari ottaaksesi yhteyttä kääntäjään"
        },
        "captions": {
          "langPair": "(* Pakollinen yhteydenottoon)"
        }
      },
      "clerkTranslatorListing": {
        "header": {
          "name": "Nimi",
          "languagePairs": "Kieliparit",
          "authorisationBasis": "Peruste",
          "valid": "Voimassa",
          "authorisationBeginDate": "Alkamispäivä",
          "authorisationEndDate": "Päätymispäivä",
          "permissionToPublish": "Julkaistu"
        },
        "detailsButton": "Lisätiedot",
        "values": { "yes": "Kyllä", "no": "Ei" }
      },
      "clerkTranslatorFilters": {
        "authorisationBasis": {
          "title": "Auktorisoinnin peruste",
          "placeholder": "Peruste"
        },
        "authorisationStatus": {
          "authorised": "Auktorisoidut",
          "expiring": "Umpeutuvat",
          "expired": "Umpeutuneet",
          "notAuthorised": "Ei auktorisoidut"
        },
        "languagePair": {
          "title": "Haku kielittäin",
          "fromPlaceholder": "Mistä",
          "toPlaceholder": "Mihin"
        },
        "name": {
          "title": "Haku nimellä",
          "placeholder": "Nimi"
        },
        "town": {
          "title": "Haku asuinkunnalla",
          "placeholder": "Asuinkunta"
        }
      },
      "clerkTranslatorOverview": {
        "buttons": {
          "save": "Tallenna",
          "back": "Takaisin",
          "next": "Seuraava",
          "cancel": "Peruuta"
        },
        "translatorDetails": {
          "header": {
            "personalInformation": "Henkilötiedot",
            "address": "Osoite",
            "contactInformation": "Yhteystiedot",
            "extraInformation": "Lisätiedot"
          },
          "buttons": {
            "edit": "Muokkaa"
          },
          "fields": {
            "lastName": "Sukunimi",
            "firstName": "Etunimet",
            "identityNumber": "Henkilötunnus",
            "street": "Katuosoite",
            "postalCode": "Postinumero",
            "town": "Postitoimipaikka",
            "country": "Maa",
            "email": "Sähköpostiosoite",
            "phoneNumber": "Puhelinnumero"
          }
        },
        "authorisations": {
          "header": "Auktorisoinnit",
          "buttons": {
            "add": "Lisää auktorisointi"
          },
          "fields": {
            "languagePair": "Kielipari",
            "startDate": "Alkamispäivä",
            "endDate": "Päätymispäivä",
            "basis": "Peruste",
            "isValid": "Voimassa",
            "isPublished": "Julkaistu",
            "diaryNumber": "Asiatunnus",
            "save": "Tallenna",
            "delete": "Poista",
            "edit": "Muokkaa"
          },
          "values": { "yes": "Kyllä", "no": "Ei" }
        },
        "notFound": "Valittua kääntäjää ei löytynyt"
      },
      "contactRequestForm": {
        "title": "Yhteydenottopyynnön jättäminen",
        "description": "Tällä yhteydenottolomakkeella voit jättää valitsemillesi kääntäjille yhteydenottopyynnön. Kirjoitathan viestiin mahdollisimman tarkan kuvauksen tarvittavasta käännöstyöstä.",
        "buttons": {
          "cancel": "Peruuta",
          "previous": "Takaisin",
          "next": "Seuraava",
          "submit": "Lähetä"
        },
        "steps": {
          "verifySelectedTranslators": "Vahvista kääntäjät",
          "fillContactDetails": "Täytä yhteystiedot",
          "writeMessage": "Kirjoita viesti",
          "previewAndSend": "Esikatsele ja lähetä",
          "done": "Valmis!"
        },
        "doneStep": {
          "title": "Yhteydenottopyyntösi on lähetetty ja sinuun ollaan yhteydessä.",
          "description": "Tämä sivu ohjaa sinut automaattisesti takaisin etusivulle. Jos mitään ei tapahdu voit klikata alla olevasta napista.",
          "button": "Etusivu"
        },
        "chosenTranslatorsForLanguagePair": "Valitut kääntäjät kielistä",
        "characters_one": "merkki",
        "characters_other": "merkkiä",
        "formLabels": {
          "firstName": "Etunimi",
          "lastName": "Sukunimi",
          "email": "Sähköpostiosoite",
          "phoneNumber": "Puhelinnumero",
          "contactInfo": "Yhteystiedot",
          "message": "Viesti",
          "writeMessageHere": "Viesti"
        },
        "errorDialog": {
          "title": "Virhe lähetettäessä yhteydenottopyyntöä.",
          "description": "Kokeile myöhemmin uudelleen. Jos ongelma toistuu, ota yhteyttä sähköpostilla: akt@oph.fi",
          "back": "Takaisin"
        },
        "cancelRequestDialog": {
          "title": "Peruuta yhteydenottopyyntö",
          "description": "Haluatko peruuttaa yhteydenottopyynnön? Menetät täyttämäsi tiedot",
          "back": "Takaisin",
          "yes": "Kyllä"
        }
      }
    },
    "pages": {
      "homepage": {
        "title": "Auktorisoitujen kääntäjien rekisteri",
        "description": "Auktorisoitujen kääntäjien tutkintolautakunta ylläpitää rekisteriä auktorisoiduista kääntäjistä. Hakukoneella voit hakea kääntäjiä joko nimen tai käännöskielen mukaan. Voit jättää valitsemellesi kääntäjille yhteydenottopyynnön alla olevalla hakulomakkeella.",
        "note": "Huomaa, että kielittäin haettaessa mahdolliset hakuvaihtoehdot ovat kotimaisista kielistä (suomi, ruotsi, saamen kielet) vieraisiin kieliin tai toisinpäin sekä kotimaisten kielten välillä.",
        "filters": {
          "title": "Hae auktorisoitua kääntäjää"
        },
        "noSearchResults": "Ei hakutuloksia",
        "requestContact": "Ota yhteyttä"
      },
      "clerkHomepage": {
        "title": "Auktorisoitujen kääntäjien rekisteri",
        "register": "Rekisteri",
        "sendEmail": "Lähetä sähköposti",
        "emptyBtn": "Tyhjennä valinnat"
      },
      "clerkSendEmailPage": {
        "title": "Lähetä sähköposti",
        "selectedCount_one": "{{count}} kääntäjä valittu",
        "selectedCount_other": "{{count}} kääntäjää valittu",
        "sections": {
          "recipients": "Vastaanottajat",
          "subject": "Aihe",
          "message": "Viesti"
        },
        "labels": {
          "subject": "Kirjoita aihe",
          "message": "Kirjoita viestisi tähän"
        },
        "buttons": {
          "cancel": "Peruuta",
          "send": "Lähetä"
        },
        "toasts": {
          "success": "Sähköpostisi lähetettiin kääntäjille.",
          "error": "Sähköpostin lähetys ei onnistunut. Virhe voi olla tilapäinen, yritä uudestaan."
        },
        "dialogs": {
          "cancel": {
            "title": "Peruuta sähköpostin lähetys",
            "description": "Haluatko peruuttaa sähköpostin lähetyksen? Menetät täyttämäsi tiedot.",
            "back": "Takaisin",
            "yes": "Kyllä"
          },
          "send": {
            "title": "Lähetä sähköposti",
            "description": "Haluatko lähettää sähköpostin? Viesti lähetetään {{count}} kääntäjälle.",
            "back": "Takaisin",
            "yes": "Kyllä"
          }
        }
      },
      "clerkTranslatorOverviewPage": {
        "title": "Kääntäjän tiedot"
      },
      "translator": {
        "name": "Nimi",
        "languagePairs": "Kieliparit",
        "town": "Asuinkunta"
      },
      "notFoundPage": {
        "title": "Valitettavasti etsimääsi sivua ei löytynyt",
        "description": "Osoite saattaa olla väärä tai vanhentunut. Voit palata etusivulle alla olevasta napista",
        "homePageBtn": "Etusivu"
      }
    },
    "errors": {
      "loadingFailed": "Tietojen lataus epäonnistui, yritä myöhemmin uudelleen.",
      "customTextField": {
        "required": "Tieto on pakollinen",
        "maxLength": "Tekstin pituus ei saa ylittää 1000 merkkiä",
        "emailFormat": "Sähköpostiosoite on virheellinen",
        "telFormat": "Puhelinnumero on virheellinen"
      }
    }
  }
}
