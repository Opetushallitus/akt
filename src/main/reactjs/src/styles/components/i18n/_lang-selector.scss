.lang-selector {
  display: flex;
  align-items: center;

  &__icon {
    margin-right: 2px;
  }

  & &__select {
    margin-top: 2px;

    &,
    & > svg {
      color: $color-text;
    }
  }
}
