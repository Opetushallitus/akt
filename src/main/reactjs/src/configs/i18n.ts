import i18n, { use, changeLanguage } from 'i18next';
import {
  initReactI18next,
  useTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import transFI from 'public/i18n/fi-FI.json';
import transSV from 'public/i18n/sv-SE.json';
import transEN from 'public/i18n/en-GB.json';
import langsFI from 'public/i18n/koodisto/langs/fi-FI.json';
import langsSV from 'public/i18n/koodisto/langs/sv-SE.json';
import langsEN from 'public/i18n/koodisto/langs/en-GB.json';

// Defaults and resources
const langFI = 'fi-FI';
const langSV = 'sv-SE';
const langEN = 'en-GB';

const detectionOptions = {
  order: ['localStorage', 'htmlTag'],
  caches: ['localStorage'],
};

const supportedLangs = [langFI, langSV, langEN];

const resources = {
  [langFI]: {
    translation: transFI,
    langs: langsFI,
  },
  [langSV]: {
    translation: transSV,
    langs: langsSV,
  },
  [langEN]: {
    translation: transEN,
    langs: langsEN,
  },
};

// Check types. It seems to be unused, but ts does some type checking with it.
declare module 'react-i18next' {
  interface CustomTypeOptions {
    defaultNS: typeof langFI;
    resources: {
      [langFI]: typeof transFI;
      [langSV]: typeof transSV;
      [langEN]: typeof transEN;
    };
  }
}

export const initI18n = () => {
  return use(initReactI18next)
    .use(LanguageDetector)
    .init({
      fallbackNS: 'langs',
      resources,
      detection: detectionOptions,
      fallbackLng: langFI,
      load: 'currentOnly',
      debug: process.env.NODE_ENV === 'development',
    });
};

export const useAppTranslation = (options: UseTranslationOptions<string>) => {
  // @ts-expect-error ts import fail
  return useTranslation(undefined, options);
};

export const getCurrentLang = (): string => {
  return i18n.language;
};

export const getSupportedLangs = (): string[] => {
  return supportedLangs;
};

export const changeLang = (language: string) => {
  return changeLanguage(language);
};

export const onLangChanged = (callback: (language: string) => void) => {
  return i18n.on('languageChanged', callback);
};
