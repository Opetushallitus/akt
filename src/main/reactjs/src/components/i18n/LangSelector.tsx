import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import {
  FormControl,
  Select,
  MenuItem,
  SelectChangeEvent,
} from '@mui/material';
import LanguageIcon from '@mui/icons-material/Language';

import { changeLang, getCurrentLang, getSupportedLangs } from 'configs/i18n';

export const LangSelector: FC = () => {
  const { t } = useTranslation();
  const [finnish, swedish, english] = getSupportedLangs();

  const handleLangChange = (event: SelectChangeEvent) => {
    changeLang(event.target.value);
  };

  return (
    <div className="lang-selector">
      <LanguageIcon className="lang-selector__icon" fontSize="small" />
      <FormControl>
        <Select
          autoWidth
          disableUnderline
          variant="standard"
          value={getCurrentLang()}
          onChange={handleLangChange}
          className="lang-selector__select"
          data-testid="lang-selector"
        >
          <MenuItem value={finnish}>
            {t('akt.component.header.lang.fi')}
          </MenuItem>
          <MenuItem value={swedish}>
            {t('akt.component.header.lang.sv')}
          </MenuItem>
          <MenuItem value={english}>
            {t('akt.component.header.lang.en')}
          </MenuItem>
        </Select>
      </FormControl>
    </div>
  );
};
