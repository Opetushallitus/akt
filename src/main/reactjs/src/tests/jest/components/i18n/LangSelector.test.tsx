import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import { LangSelector } from 'components/i18n/LangSelector';

describe('LangSelector', () => {
  it('should render LangSelector correctly', () => {
    const tree = renderer.create(<LangSelector />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should show the Finnish language as a default language', () => {
    render(<LangSelector />);

    expect(screen.getByText('akt.header.lang.fi')).toBeInTheDocument();
  });
});
