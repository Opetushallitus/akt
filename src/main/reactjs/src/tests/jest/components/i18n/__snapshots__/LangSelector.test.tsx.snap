// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangSelector should render LangSelector correctly 1`] = `
<div
  className="lang-selector"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall lang-selector__icon css-ptiqhd-MuiSvgIcon-root"
    data-testid="PublicIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
    />
  </svg>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <div
      className="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root lang-selector__select css-1aa5qj0-MuiInputBase-root-MuiInput-root"
      data-testid="lang-selector"
      onClick={[Function]}
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
         
        akt.header.lang.fi
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        disabled={false}
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value="fi-FI"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
